programs/zk-token-proof/src/lib.rs

```diff
@@ -192,5 +192,17 @@ declare_process_instruction!(process_instruction, 0, |invoke_context| {
             ic_msg!(invoke_context, "VerifyPubkeyValidity");
             process_verify_proof::<PubkeyValidityData, PubkeyValidityProofContext>(invoke_context)
         }
+        ProofInstruction::VerifyCiphertextCommitmentEquality => {
+            if native_programs_consume_cu {
```

Comment:
Ah you are right. I think it makes sense to remove it here and also for the other range proof instructions as well.

Reply:
So actually, I might be misunderstanding things, but is `native_programs_consume_cu` fully active?
It seems like it makes sense not have the check for code that are added after the feature gate is fully activated since that is a meaningless check.
But if the feature gate is only created, but not yet fully active, then it seems natural to include the check?

Reply:
I'll touch base with you about this, and feature gating generally, tomorrow. Getting late here and I want to make sure I'm reasoning properly 😅

Reply:
Oh btw, I think tomorrow is unplug day, so please take your time with the reviews! Any delay here is on me.

Reply:
I think we can remove the `native_programs_consume_cu` checks from the proof program entirely. It's totally ok that when the proof program is activated, even if `native_programs_consume_cu` has not yet activated, that the proof program consume CUs for its instructions

Reply:
Okay, you are right. The instructions are expensive, so they should always consume CU units anyway. I will remove it for this instruction here and perhaps remove the others in the benchmark PR.

Reply:
> It's totally ok that when the proof program is activated, even if native_programs_consume_cu has not yet activated, that the proof program consume CUs for its instructions

Great, that was my reasoning. The corollary that I was mulling too late last night and wanted to confirm is whether any of the proof program has been activated anywhere. If so, the new instructions themselves would need to be feature-gated.